### DISCUSSION TITLE
Is possible to search HTML/XML tag in OpenGrok

### DISCUSSION THREAD

**Discussion by sammyhk** [oracle/opengrok#3964] [General]
We have some webapp project (that is html, php or jsp files) and would like to search the usage of a given tag in those files.
We tried to type in `<html>` in Full Search field and select `XML` in Type field.
It shows the query parsing result to be: `+full:html +type:xml`, and the result set contains some irreverent records, like: `<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />` 

So is it possible to search the HTML/XML tag currently?

FYI, we are using 0.12-rc2.
Thank you for your work, this tool is really useful!

**Comment by tarzanek** [comment]
the result is fully ok - xml/html docs are parsed as plaintext and you search plain word tokens
<html>  will be brought down to html  (since < and > are not usually parsed anyways, even if you try to enclose them in "", so "<html>" it won't help, since < and > are simply not in index) 

xml analyzer could be improved in a sense that it would recognize just tags, or perhaps to go even more generic the whole "full search" could be improved now that we have more powerful lucene undernath so it indexes really "full" text ... 
right now the whole search is fast, since index has usually just words, not many special chars are indexed (have a look at PlainFullTokenizer.lex) - this I guess could be relaxed since we have faster engine and faster computers - someone however needs to do some tests if this would be feasible ...

so your Q - yes, it is possible, however you get some false positives just because you search by token, and not fulltext (don't confuse OpenGroks full search and fulltext search !!!)

**Comment by tarzanek** [comment]
short term ... I guess creating own xml tokenizers (instead of using generic plain ones) might get us where we want, not sure if we need to modify full search for every single file
also I checked and ctags do parse html, so we could get more definitions out of it as well ... 
so ... have a look how other analyzers (e.g. for java) are written, then clone FullTokenizer.lex and adjust for xml/html, hmm ?
